src/Dashboard/Prompty/PlayPrompty/PromptyConsole/PromptyConsole.js

```diff
@@ -0,0 +1,508 @@
+            async function callOpenAiTextCompletion() {
+              let res = await textCompletionApiCall(
+                "Response should be suitable for young students. Strongly refrain from using any profanity, hate, or sexual references. Following is the prompt: \n" +
+                  prompt
+              );
+              return res;
+            }
+
+            callOpenAiTextCompletion().then((res) => {
+              let obj;
+
+              let responsesData = res.data.response;
+              if (promptScaffoldMode) {
+                obj = {
+                  scaffold: true,
+                  role: roleText,
+                  context: contextText,
+                  task: taskText,
+                  iterations: responsesData,
+                };
+              } else {
+                obj = {
+                  scaffold: false,
+                  promptText: openPromptText,
+                  iterations: responsesData,
+                };
+              }
+
+              setLoader(false);
+              let currentData;
+              promptyInstanceData?.generations === undefined
+                ? (currentData = [])
+                : (currentData = promptyInstanceData?.generations);
+              saveAiResponseToFirestore([obj, ...currentData]);
+            });
+          }
+        })
+        .catch((e) => throwError(e));
+    } catch (e) {
+      throwError(e);
+    }
+  }
+
+  function throwError(e) {
+    MySwal.fire({
+      title: "Error",
+      text: e,
+      icon: "error",
+    });
+  }
+
+  function saveAiResponseToFirestore(data) {
+    const db = getFirestore();
+    let docRef = doc(
+      db,
+      "prompty",
+      props.instanceCode,
+      "instances",
+      props.identifier
+    );
+    setDoc(docRef, { generations: data }, { merge: true });
+  }
+
+  if (error) {
+    MySwal.fire({
+      title: "Error",
+      text: error,
+      icon: "error",
+    }).then(() => navigate("/dash/prompty"));
+    return null;
+  }
+
+  if (loading) return <Loader />;
+
+  const ScaffoldButton = () => {
+    return (
+      <>
+        <div className="mt-4 text-right">
+          <label className="relative inline-flex items-center cursor-pointer">
+            <input
+              type="checkbox"
+              value=""
+              className="sr-only peer"
+              checked={promptScaffoldMode}
+              onChange={() => {
+                setPromptScaffoldMode(!promptScaffoldMode);
+              }}
+            />
+            <div className="w-11 h-6 bg-gray-200 rounded-full peer peer-focus:ring-4 peer-focus:ring-slate-400 dark:peer-focus:ring-slate-400 dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-0.5 after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600"></div>
```

Comment:
[not urgent] consider making this a generic `Switch` element in `src/Generic/Switch.tsx`. That way all of us can use it (and if we want to modify the styles in the future, we can do it globally)